src/Umbraco.Web.BackOffice/Controllers/ContentController.cs

```diff
@@ -1960,6 +1960,58 @@ public IActionResult PostPublishById(int id)
         return Ok();
     }
 
+    /// <summary>
+    ///     Publishes a document with a given ID and cultures.
+    /// </summary>
+    /// <param name="model"></param>
+    /// <returns></returns>
+    /// <remarks>
+    ///     The EnsureUserPermissionForContent attribute will deny access to this method if the current user
+    ///     does not have Publish access to this node.
+    /// </remarks>
+    [Authorize(Policy = AuthorizationPolicies.ContentPermissionPublishById)]
+    public IActionResult PostPublishByIdAndCulture(PublishContent model)
+    {
+        IContent? foundContent = GetObjectFromRequest(() => _contentService.GetById(model.Id));
+
+        if (foundContent == null)
+        {
+            return HandleContentNotFound(model.Id);
+        }
+
+        var languageCount = _allLangs.Value.Count();
+        if (model.Cultures == null || !model.Cultures.Any() || model.Cultures.Length == languageCount)
+        {
+            return PostPublishById(model.Id);
+        }
+
+        var results = new Dictionary<string, PublishResult>();
+
+        foreach (var culture in model.Cultures)
+        {
+            PublishResult publishResult = _contentService.SaveAndPublish(foundContent, culture, _backofficeSecurityAccessor.BackOfficeSecurity?.GetUserId().Result ?? 0);
+            results[culture] = publishResult;
+        }
+
+
+        if (results.Any(x => x.Value.Success == false))
+        {
+            var notificationModel = new SimpleNotificationModel();
+
+            foreach (var culture in results)
+            {
+                if (culture.Value.Success == false)
```

Comment:
What I've read about this is that using LINQ usually adds some overhead instead of improving speed because it uses loops internally as well. I've done some testing in a separate environment but my experience with benchmarking is not sufficient to draw any conclusions from these tests. However I do think using the Where query here improves the readability of the code slightly and therefore makes sense to add.

Reply:
Yes indeed it's not always easy to determine this kind of metrics.
But in this case the list will be relatively limited, so performance wise we should not see a big difference and, as you said, it improves readability.